### DISCUSSION TITLE
In Slideshow mode is transparency supported?

### DISCUSSION THREAD

**Discussion by twalp** [LesFerch/DesktopPic#2] [Q&A]
Thank you for a useful program. The picture I want to display is a PNG created in Photoshop of a sign with rounded corners on a transparent background. The sign has a drop shadow.   When displayed in Slideshow mode it is shown in a white window.  Is transparency supported only in "stamped" mode?

**Comment by LesFerch** [comment]
Yes, transparency only works with stamping which directly edits the background image. The slideshow is running in a headless browser, so it's a regular window. I would have to write a slideshow module in C# to handle transparency. I might just give that a try if I can find some sample code to work from.

**Comment by LesFerch** [comment]
I found [this](https://github.com/nandemonogatari/QuickView-3.0) but it [needs work](https://github.com/nandemonogatari/QuickView-3.0/issues/8).

**Comment by twalp** [comment]
Thank you for replying and being interested!  Last night when I was
Googling for a solution and ran across your program I was being a little
lazy about searching my archives for a program that I found a decade ago.
It did what I want, which is to enable me to simply display an image,
totally chromeless, and honor transparency. It's called Frameless. Here is
a link on my Google drive
<https://drive.google.com/file/d/1FNpaFHzcuKMRu_QzEYsKEfjQGouoW-g9/view?usp=sharing>of
a zipped up folder of what I think was my final collection of materials
about the program.  I hasten to send this to you before I get engulfed in
work today, but I hope to dig back into that program and see if it still
works, and see if the guy who wrote it is still alive and kicking. Maybe
there's a new version!

Anyway, thanks again. Let me know what you think if possible.

Tim

**Reply by LesFerch** [reply]
Thanks for the pointer. I hadn't seen that one before. The current version (from 2016) can be found at:

https://www.dcmembers.com/skwire/download/frameless/

It has more options, including resizing, so it might actually work with my program, although I really should write it all in C# rather than calling an external tool. I'll consider all options.

**Comment by LesFerch** [comment]
It looks like Frameless leverages Windows screensaver feature...

C:\Programs\Frameless>trid frameless.exe

TrID/32 - File Identifier v2.24 - (C) 2003-16 By M.Pontello
Definitions found:  15721
Analyzing...

Collecting data from file: Frameless.exe
 39.4% (.EXE) Win32 Executable MS Visual C++ (generic) (31206/45/13)
 16.5% (.SCR) Windows screen saver (13097/50/3)
 13.3% (.EXE) Win64 Executable (generic) (10523/12/4)
  8.3% (.DLL) Win32 Dynamic Link Library (generic) (6578/25/2)
  6.3% (.EXE) Win16 NE executable (generic) (5038/12/1)

**Comment by twalp** [comment]
Wow what a cool tool!

Thanks for the link.

**Comment by twalp** [comment]
Ah! Too bad he (apparently) did not continue developing the
FramlessAlphaTrans version.  To me the Holy Grail was combining the ability
to show animated GIFs AND transparency.  At the time I was writing programs
using AutoIt because I could quickly grasp it and had not programmed since
writing in Z80 eons before. So I needed an executable that I could simply
call and have it display the graphics I created in Photoshop. (a primitive
GIF was in the zip)

**Reply by LesFerch** [reply]
I see that Frameless is actually an AutoHotKey package. The new version combines the two Exes. It has a "trans=alpha" command line option. Transparencies work well for still images, but not for animated GIFs. The only program I have found that handles every transparent animated GIF (and animated webp) I throw at it is [**ImageEye**](https://www.fmjsoft.com/imageeye.html#main). Transparencies AND animation. Very cool. Give it a try.

Where it falls short for me is that I want a slideshow with both transparencies and size constraints. I want all images to be displayed at roughly the same size, based on area. I have the latter covered, but not the former.

**Comment by twalp** [comment]
Sorry. I don't intend to pester but I read the last section of the forum
<https://www.donationcoder.com/forum/index.php?action=search2> and thought
you might find some of it interesting (and perhaps time saving).

**Reply by LesFerch** [reply]
That link takes me to the search page.

**Comment by twalp** [comment]
oops, sorry!  I had gone to the DC forum page about Frameless, clicked on
Last to see the most recent posts, THEN I searched for transparency in that
topic, which is when I sent you the link.  Scrolling downward through the
results I had remind myself that I was zipping back in time quite fast!
 Nonetheless, maybe you'll see something of value there.

BTW, I've got a W10 VM open on my desktop with the intention of trying
ImageEye.  Thank you for the tip.

Also, thank you for introducing me to TrID. Gotta mess with it when time
permits.
